include/glow/Optimizer/TrainingPreparation.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GLOW_OPTIMIZER_TRAININGPREPARATION_H
+#define GLOW_OPTIMIZER_TRAININGPREPARATION_H
+
+#include "glow/Base/Tensor.h"
+#include "glow/Graph/Graph.h"
+
+#include <functional>
+
+namespace glow {
+
+/// Function for \p tensor initialization.
+/// Method gets following parameters Glow Function \p F, current \p node,
+/// \p input, and fills out (optionally) \p tensor.
+using TensorInitializer = void (*)(Function *F, Node *node, NodeValue &input,
```

Comment:
Out of curiosity, why did you use a function pointer type here instead of something like `std::function<void(Function*, Node*, NodeValue&)>`?

Reply:
I did, but Roman has suggested the opposite, i.e. function vs lambda. In my opinion, lambda is more flexible, cause can carry captured objects. I can reverted back to lambda if Roman is ok with that.

Reply:
I suggested to use the function, yes.

But the type itself can and probably should be std::function. This is a different thing. You can still use a usual function (i.e. non-closure) like you have today to assign to objects of this type. And you can assign closures to it as well if necessary.